quantumdb-postgresql/src/main/java/io/quantumdb/core/schema/definitions/PostgresTypes.java

```diff
@@ -12,39 +13,55 @@
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public class PostgresTypes {
 
-	public static ColumnType from(String input) {
-		Pattern pattern = Pattern.compile("^(.+)\\(([0-9]+)\\)$");
-		Matcher match = pattern.matcher(input);
+	public static ColumnType fromString(String input) {
+		Matcher match = Pattern.compile("^(.+)\\(([0-9]+),([0-9]+)\\)$").matcher(input);
 		if (match.find()) {
 			String type = match.group(1);
-			int length = Integer.parseInt(match.group(2));
-			return from(type, length);
-		}
-		else {
-			return from(input, null);
+			int precision = Integer.parseInt(match.group(2));
+			int scale = Integer.parseInt(match.group(3));
+			return from(type.trim(), precision, scale);
+		} else {
```

Comment:
Bit of a nit-pick, but try to avoid bracket closing/opening on the same line. Bit of a styling thing TBF.